internal/pkg/source/source.go

```diff
@@ -72,7 +73,7 @@ func (a *Source) dfs(n ssa.Node) {
 	a.marked[n.(ssa.Node)] = true
 
 	if instr, ok := n.(ssa.Instruction); ok {
-		a.recordIndex(instr)
+		a.record(instr)
 	}
 
 	if n.Referrers() != nil {
```

Comment:
What if we removed referrers that are in the "past" here?
Would this do the trick? Would this read better?

Reply:
I think it could be done, but I'm not sure it would read better. We would need to create a new slice to hold the referrers we want to keep.

In fact, I think we might want to move the `n.Referrers()  != nil` check into `visitReferrers`.

Also, your comment made me realize that the code for skipping referrers that are "in the past" was placed in a weird location in `visitReferrers`, so I moved it. Do you think it's better now?

Reply:
Yes, this reads better.

You could expose a property on the `Source` type that returns referrers, and this property will be aware of the "past" and "future" concepts.

Reply:
Cool.

I think that is a good idea. I've performed the refactoring, and it definitely reads better. The `visitReferrers` function was doing too many things. The refactoring is pretty big relative to the code that really needed to change in this PR though. Let me know if you want me to extract it to its own PR.